### PR TITLE
weechat: 3.6 -> 3.7 and other stuff

### PR DESCRIPTION
**Pull request by ncfavier**
https://github.com/weechat/weechat/releases/tag/v3.7

The update was also done in https://github.com/NixOS/nixpkgs/pull/195233 so feel free to merge that one first.

### Other changes unrelated to the update

Add PHP support, but do not include it in the wrapper by default, because
- it doesn't build on Darwin,
- it doubles the closure size,
- there are no official scripts written in PHP,
- it's probably broken: the [example PHP script](https://weechat.org/files/doc/stable/weechat_scripting.en.html#register_function) from the manual fails to load. If someone cares enough they can look into it.

Enable build-time tests.

Remove `-DENABLE_JAVASCRIPT=OFF`, which was made the default upstream [a long time ago](https://github.com/weechat/weechat/commit/340d6646a6371e2f224d392c9b1b44c8a20b5074).

Add myself as a maintainer since @lovek323 is now a [retired-nixpkgs-contributor](https://github.com/orgs/NixOS/teams/retired-nixpkgs-contributors).

I also have ideas on how to fix `ENABLE_DOC`, but that's for [another PR](https://github.com/NixOS/nixpkgs/pull/195245).

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/irc/weechat/default.nix [hunk 1]
```diff
@@ -1,56 +1,63 @@
...
-      version = "3.6";
+      version = "3.7";
       pname = "weechat";
 
       hardeningEnable = [ "pie" ];
 
       src = fetchurl {
         url = "https://weechat.org/files/src/weechat-${version}.tar.bz2";
-        sha256 = "sha256-GkYN/Y4LQQr7GdSDu0ucXXM9wWPAqKD1txJXkOhJMDc=";
+        sha256 = "sha256-n5kvC//h85c4IvkrCVTz+F0DcCC5rdRkvj8W3fUPXI8=";
```
Comment (mweinelt) [inline pkgs/applications/networking/irc/weechat/default.nix:46]:
```suggestion
        hash = "sha256-n5kvC//h85c4IvkrCVTz+F0DcCC5rdRkvj8W3fUPXI8=";
```

**Comment by ncfavier** [comment]
I pushed the doc fix here just to see the rebuild count.

**Comment by ncfavier** [comment]
Moved the doc fix to https://github.com/NixOS/nixpkgs/pull/195245

File: pkgs/applications/networking/irc/weechat/default.nix [hunk 2]
```diff
@@ -85,7 +92,7 @@ let
           on https://nixos.org/nixpkgs/manual/#sec-weechat .
         '';
         license = lib.licenses.gpl3;
-        maintainers = with lib.maintainers; [ lovek323 ];
+        maintainers = with lib.maintainers; [ lovek323 ncfavier ];
```
Comment (mweinelt) [inline pkgs/applications/networking/irc/weechat/default.nix:95]: Might as well prune lovek323. Last contribution to this package in 2014.

Comment (ncfavier) [reply pkgs/applications/networking/irc/weechat/default.nix]: Done. Maybe this could be done treewide for retired contributors?

Comment (ncfavier) [reply pkgs/applications/networking/irc/weechat/default.nix]: Well I guess retired committer ≠ retired contributor necessarily.

Comment (mweinelt) [reply pkgs/applications/networking/irc/weechat/default.nix]: I think we need an RFC to define the maintainer role in general and then we can aim for treewide action. I have discussed this many times with my local peers.

**Comment by ncfavier** [comment]
Failure on Darwin seems due to https://github.com/php/php-src/issues/8533. Let's just disable PHP on Darwin.

**Comment by ncfavier** [comment]
Tests [fail](https://logs.nix.ci/?key=nixos/nixpkgs.195239&attempt_id=d23a0fe2-371d-4f7a-886a-c24ee051aa22) to build on Darwin, disabled them there.

**Comment by ncfavier** [comment]
All green!

**Review by mweinelt — APPROVED** [review]

**Pull request merged**